daemon/config/config_test.go

```diff
@@ -48,7 +58,170 @@ func TestFlattenConfig(t *testing.T) {
+			DNS:        []string{},
+			DNSOptions: []string{},
+			BridgeConfig: network.BridgeConfig{
+				Name:          "net-config-test",
+				IP:            "127.0.0.1",
+				GatewayIPv4:   "192.168.5.1",
+				FixedCIDR:     "fixed-cidr",
+				PreferredIP:   "",
+				Mtu:           1500,
+				ICC:           false,
+				IPTables:      true,
+				IPForward:     true,
+				IPMasq:        false,
+				UserlandProxy: false,
+			},
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	// Test cri configuration
+	cfg = &Config{
+		IsCriEnabled: true,
+		ListenCRI:    "unix:///tmp/test/pouch/pouchcri.sock",
+		CriConfig: criconfig.Config{
+			Listen:                 "unix:///var/run/pouchd.sock",
+			NetworkPluginBinDir:    "cni-bin-dir",
+			NetworkPluginConfDir:   "cni-conf-dir",
+			SandboxImage:           "registry.hub.docker.com/library/busybox",
+			CriVersion:             "v1alpha2",
+			StreamServerPort:       "10010",
+			StreamServerReusePort:  true,
+			CriStatsCollectPeriod:  10,
+			DisableCriStatsCollect: false,
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	// Test registry configuration
+	cfg = &Config{
+		DefaultRegistry:   "registry.hub.docker.com",
+		DefaultRegistryNS: "library",
+		RegistryService: types.RegistryServiceConfig{
+			AllowNondistributableArtifactsCIDRs: []string{
+				"::1/128",
+				"127.0.0.0/8",
+			},
+			AllowNondistributableArtifactsHostnames: []string{
+				"registry.internal.corp.example.com:3000",
+				"[2001:db8:a0b:12f0::1]:443",
+			},
+			InsecureRegistryCIDRs: []string{
+				"insecure.hub.docker.com",
+			},
+			IndexConfigs: map[string]types.IndexInfo{
+				"127.0.0.1:5000": {
+					Name: "127.0.0.1:5000",
+					Mirrors: []string{
+						"https://hub-mirror.corp.example.com:5000/",
+					},
+					Secure:   false,
+					Official: false,
+				},
+			},
+			Mirrors: []string{
+				"https://[2001:db8:a0b:12f0::1]/",
+				"https://hub-mirror.corp.example.com:5000/",
+			},
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	// Test TLS configuration
+	cfg = &Config{
+		TLS: client.TLSConfig{
+			CA:               "/path/to/.pouchcert/ca.pem",
+			Cert:             "/path/to/.pouchcert/cert.pem",
+			Key:              "/path/to/.pouchcert/key.pem",
+			VerifyRemote:     true,
+			ManagerWhiteList: "docker.alibaba.com",
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	// Test lxcfs configuration
+	cfg = &Config{
+		IsLxcfsEnabled: false,
+		LxcfsBinPath:   "/usr/local/bin/lxcfs",
+		LxcfsHome:      "/var/lib/lxcfs",
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	// Test label configuration
+	cfg = &Config{
+		Labels: []string{},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	cfg = &Config{
+		Labels: []string{
+			"a=b",
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	cfg = &Config{
+		Labels: []string{
+			"a=b",
+			"c=d",
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	cfg = &Config{
+		Labels: []string{
+			"foo=bar",
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	// Test others configuration
```

Comment:
Does this test need to cover invalid configuration case?

Reply:
Absolutely yes,  unit test need to cover all of code path, including negative path.